core/src/zisk_rom_2_asm.rs

```diff
@@ -3280,6 +3284,17 @@ impl ZiskRom2Asm {
         *code += ".section .rodata\n";
         *code += ".align 64\n";
 
+        // Safety check: Ensure the minimum program address label exists
+        //
+        // This is defensive programming for rare cases where min_program_pc has no valid
+        // instruction (non-NOP padding, data in text section).
+        // In practice with NOP padding, this check never triggers and the entrypoint
+        // is the min_program_pc
+        if rom.min_program_pc >= ROM_ADDR && !rom.sorted_pc_list.contains(&rom.min_program_pc) {
+            *code +=
+                &format!("map_pc_{:x}: \t.quad pc_{:x}\n", rom.min_program_pc, rom.min_program_pc);
+        }
```

Comment:
Note on this defensive if statement -- cc @fractasy for visibility -- we can remove it, so that the code fails early in the cases mentioned